scandium-core/src/main/java/org/eclipse/californium/scandium/DTLSConnector.java

```diff
@@ -225,6 +237,27 @@ public DTLSConnector(final DtlsConnectorConfig configuration, final SessionCache
 	}
 
 	/**
+	 * Sets the executor to use for processing records.
+	 * <p>
+	 * If this property is not set before invoking the {@linkplain #start() start method},
+	 * a new {@link StripedExecutorService} is used with a thread pool of size #(CPU cores).
+	 * <p>
+	 * In a production environment the thread pool used by the executor should be much larger
+	 * than just the number of CPU cores. This helps with performing multiple handshakes
+	 * in parallel, in particular if the key exchange requires look up of identities, e.g.
+	 * in a database or using a web service.
+	 * <p>
+	 * If this method is used to set an executor, the executor will <em>not</em> be shut down
+	 * by the {@linkplain #stop() stop method}.
+	 * 
+	 * @param executor The executor.
+	 */
+	public void setExecutor(StripedExecutorService executor) {
```

Comment:
Yes, I do. I never saw a bug because of such things in the flesh. But you should not access a field from different threads without guarding the access with the same lock or use final fields or make the field volatile. This is also the case for read access. Tell me if you think I am not right.